launch/launcher_test.go

```diff
@@ -233,15 +260,197 @@ func testLauncher(t *testing.T, when spec.G, it spec.S) {
 
 				if runtime.GOOS == "windows" {
 					h.AssertEq(t, strings.ToLower(syscallExecArgsColl[0].argv0), `c:\windows\system32\notepad.exe`)
-					h.AssertEq(t, syscallExecArgsColl[0].argv, []string{"notepad", "arg1", "arg2"})
+					h.AssertEq(t, syscallExecArgsColl[0].argv, []string{"notepad", "arg1", "arg with space"})
 				} else {
 					h.AssertEq(t, syscallExecArgsColl[0].argv0, "/bin/sh")
-					h.AssertEq(t, syscallExecArgsColl[0].argv, []string{"sh", "arg1", "arg2"})
+					h.AssertEq(t, syscallExecArgsColl[0].argv, []string{"sh", "arg1", "arg with space"})
 				}
 				h.AssertEq(t, syscallExecArgsColl[0].envv, envList)
 			})
 		})
 
+		when("a start command is NOT marked as direct", func() {
+			when("the providing buildpack has api >= 0.4", func() {
+				it.Before(func() {
+					launcher.Buildpacks = []launch.Buildpack{
+						{API: "0.4", ID: "bp.1"},
+					}
+					launcher.Processes = []launch.Process{{
+						Type:        "some-type",
+						Command:     "some-command",
+						BuildpackID: "bp.1",
+					}}
+					launcher.DefaultProcessType = "some-type"
+				})
+
+				when("there are one or more args", func() {
+					it.Before(func() {
+						launcher.Processes[0].Args = []string{"arg1", "arg with space"}
+					})
+
+					it("treats command and args as bash command tokens", func() {
+						if err := launcher.Launch("/path/to/launcher", nil); err != nil {
+							t.Fatal(err)
+						}
+
+						if len(syscallExecArgsColl) != 1 {
+							t.Fatalf("expected syscall.Exec to be called once: actual %v\n", syscallExecArgsColl)
+						}
+
+						if runtime.GOOS == "windows" {
+							h.AssertEq(t, syscallExecArgsColl[0].argv0, "cmd")
+							h.AssertEq(t, syscallExecArgsColl[0].argv, []string{
+								"cmd", "/q", "/c", `some-command`, `arg1`, "arg with space",
+							})
+						} else {
+							h.AssertEq(t, syscallExecArgsColl[0].argv0, "/bin/bash")
+							h.AssertEq(t, syscallExecArgsColl[0].argv, []string{
+								"bash", "-c",
+								`exec bash -c '"$(eval echo \"$0\")" "$(eval echo \"$1\")" "$(eval echo \"$2\")"' "${@:1}"`,
```

Comment:
I'll admit that this syntax still leaves me feeling slightly confused. Can we document in some detail how this works? 

Where I'm getting lost: there are three evals, but four "arguments" including the path to launcher. I see that `@:1` will trim off the first argument but then how does bash know what to exec? I imagine others might also have this problem...

Reply:
To summarize from elsewhere - there is an outer bash, and an inner bash. /path/to/launcher is the 0th arg to the outer bash, and some-command is the 0th arg to the inner bash (thanks to `${@:1}`).